.github/workflows/ci.yml

```diff
@@ -12,17 +12,22 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [18.x]
+        node-version: [20]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
-      - run: npm ci
-      - run: npm run build
-      - run: npm run lint
-      - run: npm test
+          cache: 'pnpm'
+      - run: pnpm install
```

Comment:
can we use `--frozen-lockfile`? to simulate the same behaviour as `npm ci` and usually used for CI pipelines to avoid versioning issues?